backend/funix/widget/builtin.py

```diff
@@ -11,96 +35,233 @@ def decorator(cls):
     return decorator
 
 
-def put_config(widget, *args, **kwargs):
-    def decorator(cls):
+def put_config(widget: callable, *args, **kwargs):
```

Comment:
how about "attach_config". It is not really a put becaues widget callable is not a dictionary so to speak.

Reply:
This sounds more reasonable, rename done!